docs/develop/01_sdk/02_cookbook/06_signCallback.md

```diff
@@ -0,0 +1,78 @@
+---
+id: signCallback
+title: SignCallback
+---
+
+import TsJsBlock from '@site/src/components/TsJsBlock';
+
+import SignCallback from '!!raw-loader!@site/code_examples/core_features/signCallback/signCallback.ts';
+import SignExtrinsicCallback from '!!raw-loader!@site/code_examples/core_features/signCallback/signExtrinsicCallback.ts';
+import GetStoreTxSignCallback from '!!raw-loader!@site/code_examples/core_features/signCallback/getStoreTxSignCallback.ts';
+
+Signing data involves using the private key and therefore needs to be secure.
+There are many different options how data could be sign.
+You might have the private key stored in memory and are therefore able to simply sign the data.
+This is the easiest option but also comes with higher security risk.
+Storing the private key on a separate device or inside a sandboxed application can increase security.
+But to enable these security options, we need a generic interface to talk to the signer.
+This is what the `SignCallback` does.
+
+The `SignCallback` defines an interface between the SDK and an arbitrary signing strategy.
+May it be a ledger, an air gaped phone or your browser extension.
+The interface is generic enough to support implementations for all these security measures.
+There are three different flavours of the `SignCallback`.
+
+## SignCallbacks Family
+
+There are three types of signing callbacks.
+The `SignCallback` is the most general and can be used in most all cases, except when a DID is registered on chain.
+The `SignExtrinsicCallback` is a special `SignCallback` which can only be used to sign extrinsics.
+Every `SignCallback` can also be used as a `SignExtrinsicCallback`.
+The `GetStoreTxSignCallback` can only be used to sign the registration of a new DID.
+
+### SignCallback
+
+The plain `SignCallback` signs arbitrary data.
```

Comment:
We have a way to generate JSON files dynamically at build time, so perhaps we could use that here, and then add text below it? So in case anything changes in the future, we can catch the change and update the docs accordingly.

Reply:
I guess also using the TS would work, perhaps.

Reply:
You want to add the call signature? I thought about that as well. I wanted to export the type definition from the SDK and even had the crazy idea to add the SDK as a git submodule or similar. 😁